### PR TITLE
Add iterators to `wirebson.Array` and `wirebson.Document`

### LINKED ISSUE #9: Add iterators and remove build tags
We should add Go 1.23-style iterators over Array values and Document field/values. See https://tip.golang.org/doc/go1.23#language

While we will not be able to use them everywhere until 1.23 is released, we could already implement and test them there. See #10.

We should remove build tags once 1.23 is used everywhere. See references to this issue.

### PR DESCRIPTION
**Pull request by chilagrow**
Closes #9.

### CODE REVIEW

File: wirebson/array_iter.go [hunk 1]
```diff
@@ -21,7 +21,16 @@
 package wirebson
 
 import (
-	_ "iter"
+	"iter"
 )
 
-// TODO https://github.com/FerretDB/wire/issues/9
+// All returns an iterator over all elements of the array.
+func (arr *Array) All() iter.Seq[any] {
```
Comment (AlekSi) [inline wirebson/array_iter.go:28]: Let's use `iter.Seq2[int, any]` for `All()` and add `Values() iter.Seq[any]` to make it compatible with `slices` package:
* https://pkg.go.dev/slices@go1.23rc2#All
* https://pkg.go.dev/slices@go1.23rc2#Values

Comment (chilagrow) [reply wirebson/array_iter.go]: That's sensible :+1:

Comment (chilagrow) [reply wirebson/array_iter.go]: For documents, do we want to add something such as `Keys() iter.Seq[any]` and `Values() iter.Seq[any]` as well?

Comment (AlekSi) [reply wirebson/array_iter.go]: We could, but I do not really see a use case yet

Comment (chilagrow) [reply wirebson/array_iter.go]: Neither, ok let's not do it for now.

**Review by AlekSi — APPROVED** [review]

File: wirebson/document.go [hunk 2]
```diff
@@ -119,11 +119,10 @@ func (doc *Document) FieldNames() []string {
 // If document contains duplicate field names, it returns the first one.
 // To get other fields, a for/range loop can be used with [Document.Len] and [Document.GetByIndex].
 // Or iterators.
-// TODO https://github.com/FerretDB/wire/issues/9
 func (doc *Document) Get(name string) any {
-	for _, f := range doc.fields {
-		if f.name == name {
-			return f.value
```
Comment (AlekSi) [inline wirebson/document.go:126]: That change should be reverted

**Pull request merged**